### PR TITLE
fix: update dependencies and fix types for Proxy

### PR DESCRIPTION
**Pull request by nolanlawson**
Updates dependencies, most notably TypeScript which has a new definition for Proxy objects which breaks some of our existing types: https://github.com/microsoft/TypeScript/pull/35594

Notably it looks like we've been assuming that `PropertyKey` (which is `string | number | symbol`) is fine to be used as proxy property keys, when actually the spec only allows `string | symbol`. So I added a `ProxyPropertyKey` type to handle this.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -40,7 +40,8 @@ function rollupConfig({ formats, prod }) {
             format,
             banner,
             footer,
-            file: path.join(targetDirectory, targetName)
+            file: path.join(targetDirectory, targetName),
+            exports: 'auto'
```
Comment (nolanlawson) [inline rollup.config.js:44]: Does not seem to change the output file format, and suppresses a warning from Rollup.

File: yarn.lock [hunk 2]
```diff
@@ -9,11 +9,23 @@
   dependencies:
     "@babel/highlight" "^7.12.13"
 
+"@babel/code-frame@^7.14.5":
```
Comment (pmdartus) [inline yarn.lock:12]: I would recommend running `yarn upgrade` to get rid of the duplicate packages.

File: rollup.config.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ function rollupConfig({ formats, prod }) {
             exclude: ['test/*'],
             typescript: require('typescript')
         }),
-        prod !== undefined && replace({ 'process.env.NODE_ENV': replaceToken }),
+        prod !== undefined && replace({ 'process.env.NODE_ENV': replaceToken, preventAssignment: true }),
         prod && terser()
     ].filter(Boolean);
```
Comment (pmdartus) [inline rollup.config.js:32]: Filtering is not needed anymore in the latest versions of rollup. 

```suggestion
    ];
```

File: rollup.config.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ function rollupConfig({ formats, prod }) {
             exclude: ['test/*'],
             typescript: require('typescript')
```
Comment (pmdartus) [inline rollup.config.js:28]: The rollup plugin now requires `typescript` to be installed as a peer dependency. It is not needed to pass typescript anymore.

```suggestion
```

**Review by pmdartus — APPROVED** [review]
Minor nitpicks otherwise LGTM.

File: yarn.lock (hunk 2 above)
Comment (nolanlawson) [reply yarn.lock:12]: Done!

**Pull request merged**